### ISSUE TITLE
apoc.export.csv.query() config for quotes: 'ifNeeded' not working

### ISSUE THREAD

**Issue by JMHReif** [neo4j-contrib/neo4j-apoc-procedures#4014]
## Expected Behavior (Mandatory)
Should produce a CSV that only has quotes around the string values.

## Actual Behavior (Mandatory)
Not having any quotes around any value. Output looks the same as when quotes: 'none'

## How to Reproduce the Problem
- Test the apoc.export.csv.query quote issue with config `stream: true, quotes: 'ifNeeded'`

### Simple Dataset (where it's possibile)

```
//Insert here a set of Cypher statements that helps us to reproduce the problem
WITH ["Whole Bean/Teas", "Beverages", "Add-ons", "Food", "Merchandise"] as list
UNWIND list as item
MERGE (g:Group {group: item});
```

### Steps (Mandatory)

  1. Create the sample data set
  2. Run the following query:
```
WITH "MATCH (group:Group) 
WITH collect(group) AS groups 
WITH groups, range(0, size(groups) - 1) AS ordinal_positions
UNWIND ordinal_positions AS id
WITH id+1 as group_id, groups[id] as group
return group_id, group.group as product_group" AS query
CALL apoc.export.csv.query(query, null, {stream: true, quotes: 'ifNeeded'})
YIELD data
RETURN data;
```

## Screenshots (where it's possibile)
![Screenshot 2024-03-14 at 4 01 35 PM](https://github.com/neo4j-contrib/neo4j-apoc-procedures/assets/14850786/813c75b6-e500-4ba2-87dc-4fb56b935d2b)

## Specifications (Mandatory)

Currently used versions

### Versions

  - OS:
  - Neo4j: Aura
  - Neo4j-Apoc: Aura

**Comment by JoelBergstrand** [comment]
Thanks for the excellent bug report! Agree that it looks like a bug, we'll look into it 🧐

**Comment by JoelBergstrand** [comment]
I've looked into it now and this is the expected behavior of using `'ifNeeded'`. It escapes characters such as commas and not the column as a whole. Alternatively, you could use `'always'` which puts all values in quotation marks and would get you the following result: 

``` 
"group_id","product_group" 
"1","Whole Bean/Teas"
"2","Beverages"
...
```

**Comment by JMHReif** [comment]
That turns everything into strings, including integer values, dates, and so on. I was expecting the 'if-needed' option to not turn integers and other non-string data types into strings. 😕

**Comment by gem-neo4j** [comment]
Yeah, like Joel said, the 'ifNeeded' option is about adding quotes around characters that have special characters, this is mainly so it doesn't break the csv file or a reader, reading in the data. In a csv file, APOC is expecting that a non-quoted string will still work as a string.

I assume your example is just an example, but one way to fix it with Cypher would be to add the quotes before giving it to APOC:

```
WITH "MATCH (group:Group) 
WITH collect(group) AS groups 
WITH groups, range(0, size(groups) - 1) AS ordinal_positions
UNWIND ordinal_positions AS id
WITH id+1 as group_id, groups[id] as group
return group_id, '\"' + group.group + '\"' as product_group" AS query
CALL apoc.export.csv.query(query, null, {stream: true, quotes: 'none', differentiateNulls: true})
YIELD data
RETURN data;
```

I'll close this one as it is expected behaviour, if you want to request this as a feature let me know, and I can add a card about it, but keep in mind, we aren't actively developing APOC core atm, so there is little time spent on new features.
